doc/man3/DEFINE_STACK_OF.pod

```diff
@@ -239,10 +240,14 @@ OPENSSL_sk_delete(), OPENSSL_sk_delete_ptr(), OPENSSL_sk_dup(),
 OPENSSL_sk_find(), OPENSSL_sk_find_ex(), OPENSSL_sk_find_all(),
 OPENSSL_sk_free(), OPENSSL_sk_insert(), OPENSSL_sk_is_sorted(),
 OPENSSL_sk_new(), OPENSSL_sk_new_null(), OPENSSL_sk_new_reserve(),
-OPENSSL_sk_num(), OPENSSL_sk_pop(), OPENSSL_sk_pop_free(), OPENSSL_sk_push(),
-OPENSSL_sk_reserve(), OPENSSL_sk_set(), OPENSSL_sk_set_cmp_func(),
-OPENSSL_sk_shift(), OPENSSL_sk_sort(), OPENSSL_sk_unshift(),
-OPENSSL_sk_value(), OPENSSL_sk_zero().
+OPENSSL_sk_num(), OPENSSL_sk_pop(), OPENSSL_sk_pop_free(),
+OPENSSL_sk_pop_free_thunk(), OPENSSL_sk_push(), OPENSSL_sk_reserve(),
+OPENSSL_sk_set(), OPENSSL_sk_set_cmp_func(), OPENSSL_sk_set_thunks(),
+OPENSSL_sk_shift(), OPENSSL_sk_sort(), OPENSSL_sk_unshift(), OPENSSL_sk_value(),
+OPENSSL_sk_zero().
+
+OPENSSL_sk_set_thunks() and OPENSSL_sk_pop_free_thunk(), while public by
+necessity, are actually internal functions and should not be used.
```

Comment:
Really?  You aren't using them anywhere where they'd have to be exported publicly.

Why exactly must these be public?

Reply:
It is used in safestack.h.in which is public. Or am I missing something?

Reply:
I'm not seeing anything obvious there that couldn't be lowered in the stack.c file & completely hidden from the user.

Reply:
The problem is that you cannot infer the actual type from stack.c. But maybe I am missing something. Can you be more concrete on how it can be achieved?

Reply:
I must be missing something.  I don't see why the type is required.  We're dealing with `void *` pointers which are type less.  The only places that are casting could accept a `void *` pointer instead.

Instead of having functions which are public but documented as internal, it would be far better to hide all of this from the user.  Yes, we'd lose some type information but we're doing that for everything else in stack.c.

Even those the APIs are documented as internal only, they **will** be used and the project will have to support them essentially forever.

Reply:
Yeah, this cannot be done without at least one new public API call.

Reply:
Thanks for the explanation.